frontend/lib/immediates/complex-support.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "complex-support.h"
+
+#include <complex.h>
+
+static double complex makeDoubleComplex(double re, double im) {
+  // Some test environments don't have a working CMPLXF
+  // so use another way to set if needed.
+#if defined(CMPLX)
+  return CMPLX(re, im);
+#else
+#ifndef CHPL_DONT_USE_CMPLX_PTR_ALIASING
+#define cmplx_re64(c) (((double *)&(c))[0])
+#define cmplx_im64(c) (((double *)&(c))[1])
+  double complex val;
+  cmplx_re64(val) = re;
+  cmplx_im64(val) = im;
```

Comment:
At this point, I would do away with the `#define`s and simply include the original expressions + a comment ("set the real part" .e.g.).
```suggestion
  double complex val;
  (((double *)&val)[0]) = re; /* set the real part of the complex number */
  (((double *)&val)[1]) = im; /* set the imaginary part of the complex number */
```

Reply:
Since this code is pretty much the same as `_chpl_complex128` etc that Brad added to `chpltypes.h` in the runtime in #24184 / #24211 I wanted to follow the same pattern. I agree that the macro is not strictly necessary here, but it is arguably valuable to communicate intent -- which you handled with the comments.

@bradcray how would you feel about removing the macros in both of these places?

Reply:
As someone who habitually doesn't read comments (like, I didn't even see them here until I saw Michael's "which you handled with the comments" and prefers self-descriptive code, I'd probably still stick with the macros, personally.  But I'm admittedly less invested in the frontend/ directory than I am the runtime, so don't have a preference for this PR.  Just not sure I'd remove them from the runtime for the sake of consistency.